src/daft-plan/src/planner.rs

```diff
@@ -487,6 +487,34 @@ pub fn plan(logical_plan: &LogicalPlan, cfg: Arc<DaftConfig>) -> DaftResult<Phys
         }) => {
             let mut left_physical = plan(left, cfg.clone())?;
             let mut right_physical = plan(right, cfg.clone())?;
+
+            // If either the left or right side of the join are very small tables, perform a broadcast join with the
```

Comment:
as discussed we should not do a broadcast join if don't have to repartition the large table (it's already hash partitioned by the join key). In that case, we should repartition the smaller table to match the larger table

Reply:
Completed in the second-to-last commit just pushed!